src/Microsoft.Identity.Client/Internal/Cache/TokenCacheKeyBase.cs

```diff
@@ -0,0 +1,49 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Identity.Client.Internal.Cache
+{
+    internal abstract class TokenCacheKeyBase
+    {
+        public TokenCacheKeyBase(string clientId, string userIdentifier)
+        {
+            ClientId = clientId;
+            UserIdentifier = userIdentifier;
+        }
+
+        public string ClientId { get; set; }
+
+        public string UserIdentifier { get; set; }
+
+        public bool Equals(string string1, string string2)
```

Comment:
Method and parameter names are important. _Equals_ gives no clue about the what it should be used to compare - client ids, authorities, environments? Put yourself in the position of someone else maintaining the code in three years time: how are they going to know what this method is for?